CMakeLists.txt

```diff
@@ -70,19 +70,32 @@ project(Daemon C CXX)
 ################################################################################
 
 if(NOT DAEMON_EXTERNAL_APP)
-    option(BUILD_CLIENT "Build Daemon client" 1)
-    option(BUILD_SERVER "Build Daemonded server" 1)
-    option(BUILD_GAME_NATIVE_EXE "Build native executable game logic" 1)
-    option(BUILD_GAME_NATIVE_DLL "Build native shared library game logic" 1)
+    option(BUILD_CLIENT "Engine client executable, required to play" ON)
+    option(BUILD_SERVER "Engine server executable, required to host servers" ON)
+
+    ## About the different ways to host/play games:
+    ## Native DLL: no sandboxing, no cleaning up but debugger support. Use for dev.
+    ## NaCl exe: sandboxing, no leaks, slightly slower, hard to debug. Use for regular players.
+    ## Native exe: no sandboxing, no leaks, hard to debug. Might be used by server owners for perf.
+    ## see VirtualMachine.h for code
+
+    # can be loaded by daemon with vm.[sc]game.type 0 or 1
+    # TODO (maybe): those could probably be merged as a single option "BUILD_DISTRIBUTED_FILES"
+    option(BUILD_GAME_NACL "Build the «pexe» files, required to build the «nexe» files which are required to host servers." ON)
+    option(BUILD_GAME_NACL_NEXE "Build the «nexe» files from the «pexe» files, required to host servers." ON)
+
+    # can be loaded by daemon with vm.[sc]game.type 3
+    option(BUILD_GAME_NATIVE_DLL "Build the shared library files, mostly useful for debugging changes locally." ON)
+
+    # can be loaded by daemon with vm.[sc]game.type 2
+    option(BUILD_GAME_NATIVE_EXE "Build «native executable», which might be used for better performances by server owners" ON)
+
     if (CMAKE_SYSTEM_NAME STREQUAL CMAKE_HOST_SYSTEM_NAME)
-        option(BUILD_GAME_NACL "Build PNaCl game logic" 1)
-        option(BUILD_GAME_NACL_NEXE "Build NaCl game logic" 1)
-    else()
-        set(BUILD_GAME_NACL 0)
-        set(BUILD_GAME_NACL_NEXE 0)
+        option(BUILD_GAME_NACL ON)
+        option(BUILD_GAME_NACL_NEXE ON)
     endif()
-    option(BUILD_TTY_CLIENT "Build Daemon headless client" 1)
-    option(BUILD_DUMMY_APP "Build the dummy app for daemon" 0)
+    option(BUILD_TTY_CLIENT "Engine «client» with no graphical display, used to control servers without logging in" ON)
```

Comment:
Remove "used to control servers without logging in" - there are multiple use cases.

Reply:
Are there others? This is what I supposed, I'd like to know if there are other uses so that the wording can be improved.

Reply:
It's useful in a way similar to the dummy app, having binaries with incrementally larger portions of functionality included.

It can be used to script some debugging/testing scenarios.